src/CamelKNodeProvider.ts

```diff
@@ -62,7 +62,7 @@ export class CamelKNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 			}
 			return Promise.resolve(oldNodes);
 		}
-		return Promise.reject();
+		return Promise.reject(undefined);
```

Comment:
what is the advantage of rejecting with undefined compared to rejecting with empty method?

Shouldn't we try to reject with an Error?

Reply:
Yes, rejecting with an error is probably best. Though it is actually making no difference in the result. Rejects will be handled in the catch block and resolves will be handled in the then block. If the catch block is not taking care of the error then it is not important what you put into the reject. From what I see in the comment that function is only used in the tests, but I haven't double checked yet.

Reply:
Checked and its used internal in the CamelKNodeProvider and in a test. The positions inside CamelKNodeProvider where it is used don't look at the reject cause. I think that whole class needs some refactor at some point but that was not the subject for this PR. If you are fine with it I would like to defer it to a separate PR just refactoring this file.

Reply:
so maybe we can keep promise.reject() instead of introducing Promise.reject(undefined) which is bringing nothing more (or I misunderstood).
And ok to create another issue to improve this part with better error handling

Reply:
I am fine with that. Will revert to empty reject in both positions.

Reply:
Now I see why I added the parameter. Code complains that it should have a parameter. Seems to come from the linter. Anyways it builds and runs fine, so I will keep it without parameter for now until we refactor that file.